### PR TITLE
Navigation Improvements

### PR DESCRIPTION
**Pull request by theimpulson**

### CODE REVIEW

**Comment by grote** [comment]
i see something strange happening here when navigating back from fragments. do you see the same?


https://github.com/user-attachments/assets/a315369c-0524-43da-8701-61dd43d0a6d2

Note the toolbar with title and overflow menu.

**Comment by theimpulson** [comment]
> i see something strange happening here when navigating back from fragments. do you see the same?

Seems fine on Android 15 here

File: storage/demo/src/main/java/de/grobox/storagebackuptester/restore/DemoSnapshotFragment.kt [hunk 1]
```diff
@@ -31,7 +31,7 @@ class DemoSnapshotFragment : SnapshotFragment() {
         bottomStub.layoutResource = R.layout.footer_snapshot
         val footer = bottomStub.inflate()
         footer.findViewById<Button>(R.id.button).setOnClickListener {
-            requireActivity().onBackPressed()
+            requireActivity().onBackPressedDispatcher.onBackPressed()
```
Comment (grote) [inline storage/demo/src/main/java/de/grobox/storagebackuptester/restore/DemoSnapshotFragment.kt:34]: If you change this for the demo app, don't you need to add `android:enableOnBackInvokedCallback="true"` to its manifest as well?

Comment (theimpulson) [reply storage/demo/src/main/java/de/grobox/storagebackuptester/restore/DemoSnapshotFragment.kt:34]: Done

**Comment by grote** [comment]
Still seeing the same thing on Android 15 here.

**Pull request closed without merging**